### PR TITLE
ENT-4612: policy calls ensure_feeders(ARRAY) to ensure feeders in database

### PR DESCRIPTION
**Pull request by craigcomstock**
Changelog: none

merge with https://github.com/cfengine/nova/pull/1437

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @nickanderson can review this?

File: cfe_internal/enterprise/federation/federation.cf [hunk 1]
```diff
@@ -56,6 +56,8 @@ bundle agent config
         # To ensure we are using a remote hub that's enabled
         if => strcmp( "on", "$(data[remote_hubs][$(remotes)][target_state])" );
       "fingerprints" slist => getvalues( fingerprint );
+      "feeder[$(remotes)]" string => "$(data[remote_hubs][$(remotes)][ui_name])",
```
Comment (nickanderson) [inline cfe_internal/enterprise/federation/federation.cf]: ui_name, isnt this changing to public key sha now?

File: cfe_internal/enterprise/federation/federation.cf [hunk 2]
```diff
@@ -397,12 +399,33 @@ bundle agent superhub_schema
       "$(cfengine_enterprise_federation:config.bin_dir)/psql_wrapper.sh"
         arglist => {
                      "cfdb",
-                     "'select superhub_schema();'",
+                     "\"select superhub_schema('$(sys.key_digest)');\"",
```
Comment (nickanderson) [inline cfe_internal/enterprise/federation/federation.cf]: I personally dislike escaping strings. So, if you also dislike escaping strings, then you can change your outer double quotes to backticks ``` https://docs.cfengine.com/docs/3.12/reference-language-concepts-variables.html#quoting

File: cfe_internal/enterprise/federation/federation.cf [hunk 3]
```diff
@@ -399,10 +401,31 @@ bundle agent superhub_schema
...
+                                            getvalues( "cfengine_enterprise_federation:config.feeder" )
+                                      ),
+                                      "']");
+
+  commands:
+    am_superhub::
+      "$(cfengine_enterprise_federation:config.bin_dir)/psql_wrapper.sh"
+        arglist => {
+                     "cfdb",
+                     "\"select ensure_feeders($(feeders_arg));\""
```
Comment (nickanderson) [inline cfe_internal/enterprise/federation/federation.cf]: I hate having to escape quotes. Consider using single quote or backtick on the outside, then you don't have to https://docs.cfengine.com/docs/3.12/reference-language-concepts-variables.html#quoting

File: cfe_internal/enterprise/federation/federation.cf (hunk 1 above)
Comment (craigcomstock) [reply cfe_internal/enterprise/federation/federation.cf]: yep. awesome. fixing.

**Pull request merged**